### PR TITLE
✨ Extended BDL input iterator to support different SiDB input representations

### PR DESCRIPTION
**Pull request by Drewniok**
## Description

This PR extends the _BDL input iterator_ to accommodate the varying ways input information can be implemented in SiDB technology. In some cases, such as with Bestagon gates, the binary value `0` may be represented by a perturber placed farther away compared to the `1` state. In contrast, _SiQAD_ gates typically represent `0` by the absence of a perturber and `1` by the presence of a perturber.

In addition, this PR includes the missing experiment scripts needed for temperature simulation.

## Checklist:

- [x] The pull request only contains commits that are related to it.
- [ ] I have added appropriate tests and documentation.
- [x] I have added a changelog entry.
- [x] I have created/adjusted the Python bindings for any new or updated functionality.
- [x] I have made sure that all CI jobs on GitHub pass.
- [ ] The pull request introduces no new warnings and follows the project's style guidelines.

### CODE REVIEW

File: include/fiction/algorithms/simulation/sidb/critical_temperature.hpp [hunk 1]
```diff
@@ -149,13 +143,13 @@ class critical_temperature_impl
             layout{lyt},
             params{ps},
             stats{st},
-            bii(bdl_input_iterator<Lyt>{layout, params.bdl_params})
+            bii(bdl_input_iterator<Lyt>{layout, params.input_bdl_iterator_params}),
+            critical_temperature{ps.max_temperature}
```
[the remark this answers is not in the document]
Comment (marcelwa) [reply include/fiction/algorithms/simulation/sidb/critical_temperature.hpp:148]: @Drewniok this is actually meaningful.

File: cli/cmd/simulation/temp.hpp [hunk 2]
```diff
@@ -198,6 +197,10 @@ class temp_command : public command
      * Critical temperature statistics.
      */
     fiction::critical_temperature_stats stats{};
+    /**
+     * Critical temperature.
+     */
+    double ct = 0.0;
```
Comment (marcelwa) [inline cli/cmd/simulation/temp.hpp:203]: Maybe you want to reset this value in the beginning of the `execute` function

File: include/fiction/algorithms/iter/bdl_input_iterator.hpp [hunk 3]
```diff
@@ -17,6 +17,43 @@
...
+struct bdl_input_iterator_params
+{
+    /**
+     * This enum defines how BDL inputs are manipulated within the algorithm.
+     * According to existing literature, there are two main approaches for handling BDL inputs:
+     *
+     * A perturber is used to set the input to `1`, while the absence of a perturber represents a `0`
+     *    (as described in https://ieeexplore.ieee.org/abstract/document/8963859).
+     * An input of `1` is generated by placing a perturber closer to the target, whereas a `0` is produced
+     *    by positioning the perturber farther away (as described in https://dl.acm.org/doi/10.1145/3489517.3530525).
```
Comment (marcelwa) [inline include/fiction/algorithms/iter/bdl_input_iterator.hpp:32]:
```suggestion
     * 1. A perturber is used to set the input to `1`, while the absence of a perturber represents a `0`
     *    (as described in https://ieeexplore.ieee.org/abstract/document/8963859).
     * 2. An input of `1` is generated by placing a perturber closer to the target, whereas a `0` is produced
     *    by positioning the perturber farther away (as described in https://dl.acm.org/doi/10.1145/3489517.3530525).
```

File: include/fiction/algorithms/simulation/sidb/random_sidb_layout_generator.hpp [hunk 4]
```diff
@@ -73,7 +73,7 @@ struct generate_random_sidb_layout_params
      * even impossible to find several unique (given by number_of_unique_generated_layouts) layouts. Therefore, this
      * parameter sets a limit for the maximum number of tries.
      */
-    uint64_t maximal_attempts_for_multiple_layouts = 10E6;
+    uint64_t maximal_attempts_for_multiple_layouts = 1000000;
```
Comment (marcelwa) [inline include/fiction/algorithms/simulation/sidb/random_sidb_layout_generator.hpp:76]:
```suggestion
    uint64_t maximal_attempts_for_multiple_layouts = 1'000'000;
```

**Comment by marcelwa** [comment]
@Drewniok can you still add a CHANGELOG entry, please?

**Comment by Drewniok** [comment]
> @Drewniok can you still add a CHANGELOG entry, please?

Done!

File: docs/changelog.rst [hunk 5]
```diff
@@ -5,6 +5,16 @@ All notable changes to this project will be documented in this file.
 
 The format is based on `Keep a Changelog <https://keepachangelog.com/en/1.0.0/>`_.
 
+v0.6.5 - 2024-09-06
```
Comment (marcelwa) [inline docs/changelog.rst:8]:
```suggestion
Unreleased```

**Review by marcelwa — CHANGES_REQUESTED** [review]

**Review by marcelwa — APPROVED** [review]

**Pull request merged**